pkg/apis/monitoring/v1alpha1/prometheusagent_types.go

```diff
@@ -91,5 +91,14 @@ func (l *PrometheusAgentList) DeepCopyObject() runtime.Object {
 // https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
 // +k8s:openapi-gen=true
 type PrometheusAgentSpec struct {
+	// Mode defines how the Prometheus operator deploys the PrometheusAgent pod(s).
+	// For now this field has no effect.
+	//
+	// (Alpha) Using this field requires the `PrometheusAgentDaemonSet` feature gate to be enabled.
+	//
+	// +kubebuilder:validation:Enum=StatefulSet;DaemonSet
+	// +optional
+	Mode *string `json:"mode,omitempty"`
+
 	monitoringv1.CommonPrometheusFields `json:",inline"`
```

Comment:
stylistic suggestion: I'd put the inline struct first.

```suggestion
	monitoringv1.CommonPrometheusFields `json:",inline"`

	// Mode defines how the Prometheus operator deploys the PrometheusAgent pod(s).
	// For now this field has no effect.
	//
	// (Alpha) Using this field requires the `PrometheusAgentDaemonSet` feature gate to be enabled.
	//
	// +kubebuilder:validation:Enum=StatefulSet;DaemonSet
	// +optional
	Mode *string `json:"mode,omitempty"`
```

Reply:
This makes sense. But I don't know if changing the order of fields in the codebase also changes the order of fields in the doc? I think it's best for users to see the `mode` field before the fields of the modes.

Reply:
I just checked the doc. Turned out the order of fields in doc doesn't follow the order in code, but the alphabet... I'll change the order in the code as you suggested then. But eventually I think it would be best to somehow show `mode` at the top. @ArthurSens Is it possible to do that in the doc? (asking you since you're working on doc and website)

Reply:
cc @AshwinSriram11 :)

Reply:
I don't know, we need to investigate 😬

Reply:
Currently `mode` is showing at the top(on the website). I will check how to change the order.

Reply:
Oh, really? Then it's good. That's what we want. No need to do anything more.

> I just checked the doc. Turned out the order of fields in doc doesn't follow the order in code, but the alphabet...

Scratch this. I checked this in `bundle.yaml`, while I should have checked `api.md` instead. Sorry for that :)